code-of-conduct.md

```diff
@@ -0,0 +1,79 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as
+contributors and maintainers pledge to making participation in our project and
+our community a harassment-free experience for everyone, regardless of age, body
+size, disability, ethnicity, sex characteristics, gender identity and expression,
+level of experience, education, socio-economic status, nationality, personal
+appearance, race, religion, or sexual identity and orientation. This list is
+not meant to be exhaustive, but merely to enumerate some common kinds of
+behaviors.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment
+include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
+  advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic
+  address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable
+behavior and are expected to take appropriate and fair corrective action in
+response to any instances of unacceptable behavior.
+
+Project maintainers have the right and responsibility to remove, edit, or
```

Comment:
So in the end, we stick with the responsibility? We had a discussion on _didn't sign up for this responsibility_. Should we remove responsability word?

I'm not sure, and I'm happy to begin with this.

Reply:
We can change this.  What would we like to put here instead?

Reply:
In my point of view, we have the responsibility to signal unacceptable behavior, but maybe not more. However, we have the right and authority to perform any of the described actions.

Reply:
Yeah, I have to agree with this concern. It has also been brought up to me by people internally when we looked at the Contributor Code of Conduct previously. Worry that encouraging this sort of behavior might only make an incident devolve more quickly.

Projects like NumPy and SciPy have a Code of Conduct Committee which determines how to act if an issue is raised to them. After which they come to some determination, which they share publicly explaining what they have decided. Would personally feel more comfortable with some group like that for Dask.

ref: https://docs.scipy.org/doc/scipy/reference/dev/conduct/code_of_conduct.html#reporting-guidelines
ref: https://docs.scipy.org/doc/scipy/reference/dev/conduct/code_of_conduct.html#incident-reporting-resolution-code-of-conduct-enforcement

Reply:
Is there a consensus here to remove the responsibility part as @guillaumeeb proposed? That is, re-word "Project maintainers have the right and responsibility to remove..." to "Project maintainers have the right to remove..."

@jakirkham I'm also in favor of the Code of Conduct Committee approach